client/updates/zypper.go

```
@@ -0,0 +1,200 @@
+
+		result = append(result, zypperUpdate{
+			name:    strings.TrimSpace(parts[2]),
+			version: strings.TrimSpace(parts[4]),
+			arch:    strings.TrimSpace(parts[5]),
+		})
+	}
+
+	return result, nil
+}
+
+type zypperInfo struct {
+	isSecurity  bool
+	needsReboot bool
+}
+
+type zypperUpdate struct {
+	name    string
+	arch    string
+	version string
+}
+
+func (p *ZypperPackageManager) getSecurityAndRebootInfo(ctx context.Context) (map[string]zypperInfo, error) {
+	output, err := p.runner.Run(ctx, p.listPatchesCmd...)
+	if err != nil {
+		return nil, err
+	}
+
+	result := make(map[string]zypperInfo)
+	for _, line := range strings.Split(output, "\n") {
+		if strings.HasPrefix(line, "Repository") {
+			continue
+		}
+
+		parts := strings.Split(line, "|")
+		if len(parts) != 7 {
+			continue
+		}
+
+		name := strings.TrimSpace(parts[1])
+		category := strings.TrimSpace(parts[2])
+		interactive := strings.TrimSpace(parts[4])
+
+		updates, err := p.listUpdatesForPatch(ctx, name)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, update := range updates {
+			result[update] = zypperInfo{
+				isSecurity:  category == "security",
+				needsReboot: interactive == "reboot",
+			}
+		}
+	}
+
+	return result, nil
+}
+
+func (p *ZypperPackageManager) listUpdatesForPatch(ctx context.Context, patchName string) ([]string, error) {
+	fullCmd := append(p.patchInfoCmd, patchName)
+	output, err := p.runner.Run(ctx, fullCmd...)
+	if err != nil {
+		return nil, err
+	}
+
+	result := []string{}
+	skipUntilConflicts := true
+	for _, line := range strings.Split(output, "\n") {
+		if skipUntilConflicts {
+			if strings.HasPrefix(line, "Conflicts") {
+				skipUntilConflicts = false
+			}
+			continue
+		}
+
+		parts := strings.Fields(line)
+		if len(parts) != 3 {
+			continue
+		}
+
+		result = append(result, parts[0])
+	}
+
+	return result, nil
+}
+
+func (p *ZypperPackageManager) checkRebootRequired(ctx context.Context) bool {
+	output, _ := p.runner.Run(ctx, p.needsRebootCmd...)
```

nit: please add an explanatory comment why the error is not handled.

---

Done